tuning/data/data_processors.py

```diff
@@ -81,33 +81,102 @@ def load_dataset(
         if (not datafile) and (not datasetconfig):
             raise ValueError("Either datafile or datasetconfig must be set")
 
+        def _load_dataset(path=None, builder=None, data_files=None, data_dir=None):
+            """
+            Helper function to load a dataset using datasets.load_dataset
+            with standardized exception handling.
+
+            Args:
+                path: The path argument for load_dataset (could be a directory, file, builder, etc.)
+                builder: Optional builder to use if provided.
+                data_files: Optional data_files list if loading from files.
+                data_dir: Optional data_dir if loading from a directory with a builder.
+            Returns: dataset
+            """
+
+            load_kwargs = {**kwargs, "split": splitName}
+            if data_dir is not None:
+                load_kwargs["data_dir"] = data_dir
+            if data_files is not None:
+                load_kwargs["data_files"] = data_files
+
+            # Determine the `path` parameter for load_dataset
+            load_path = builder if builder else path
+
+            try:
+                return datasets.load_dataset(path=load_path, **load_kwargs)
+            except DatasetNotFoundError as e:
+                # Reraise with a more context-specific message if needed
+                raise e
+            except FileNotFoundError as e:
+                # Handle file/directory not found
+                context = f"builder {builder}" if builder else f"path {path}"
+                raise ValueError(f"Data loading failed: invalid {context}.") from e
```

Comment:
for the case of builder shouldn't the context still include data path?

Reply:
Make sense.